sdks/python/setup.py

```diff
@@ -115,8 +115,7 @@ def get_version():
     'mock>=1.0.1,<3.0.0',
     'pymongo>=3.8.0,<4.0.0',
     'oauth2client>=2.0.1,<4',
-    # grpcio 1.8.1 and above requires protobuf 3.5.0.post1.
-    'protobuf>=3.5.0.post1,<4',
+    'protobuf>=3.8.0.post1,<4',
```

Comment:
> @aaltay do you think this might be an issue?
Upping the lowerbound for protobuf should be ok. @TheNeuralBit what is the reason for this change? Does something require this newer version?

Reply:
I bumped it because I used a 3.8.0 feature without realizing it (using `schema_pb2.AtomicType.STRING`, rather than `schema_pb2.STRING`). This is actually the bug I was referencing in https://lists.apache.org/thread.html/e174967bab6c224db40f66af74e7d07466812cc6b78189e55da88cb0@%3Cdev.beam.apache.org%3E

If it's going to be a problem I can change it to back and update my code, bumping it to 3.8.0 was just the lazy fix

Reply:
If it would not be a difficult fix, or would not cause future issues, it is better to keep the version range as is.

Reply:
:+1: done

I also added a dependency on numpy. It shouldn't change anything since we already had the dependency transitively through pyarrow (and perhaps others?)

Is that ok?

Reply:
Re: numpy

That is ok. It needs to be an explicit dependency at this point really.

you can remove numpy dependency under test packages and please try to keep the same range.

Reply:
Sounds good, I just moved the numpy dependency from test to required with the same range. Thanks!